src/Microsoft.CodeAnalysis.NetAnalyzers/src/Microsoft.CodeAnalysis.CSharp.NetAnalyzers/Microsoft.NetCore.Analyzers/Runtime/CSharpDetectPreviewFeatureAnalyzer.cs

```diff
@@ -3,15 +3,29 @@
 using System.Collections.Immutable;
 using System.Diagnostics.CodeAnalysis;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
 using Microsoft.NetCore.Analyzers.Runtime;
+using Analyzer.Utilities.Lightup;
 
 namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     public class CSharpDetectPreviewFeatureAnalyzer : DetectPreviewFeatureAnalyzer
     {
+        protected override ISymbol? SymbolFromAwaitOperation(IAwaitOperation operation)
+        {
+            if (operation.Syntax is not AwaitExpressionSyntax awaitSyntax)
```

Comment:
Consider testing with `await using` and `await foreach`

Reply:
These are going to need new public APIs. I'll open a followup.